### PR TITLE
Add a timeout parameter to the jenkins_trigger_build script

### PR DESCRIPTION
**Pull request by cpennington**

### CODE REVIEW

File: tubular/jenkins.py [hunk 1]
```diff
@@ -12,10 +13,49 @@
...
+    # The Build object was the only parameter to the original method call,
+    # and so it's the first and only item in the args.
+    build = orig_args[0]
+    msg = u'Timed out waiting for build {} to finish.'.format(build.name)
+    raise BackendError(msg)
+
+
+def _backoff_timeout(timeout, base=2, factor=1):
+    u"""Return a tuple of (wait_gen, max_tries) so that backoff will only try up to `timeout` seconds."""
+    # Total duration of sum(factor * base ** n for n in range(K)) = factor*(base**K - 1)/(base - 1)
```
Comment (doctoryes) [inline tubular/jenkins.py]: Maybe explain what `K` is here before the equation (if not also `base` and `factor`).

File: tubular/jenkins.py [hunk 2]
```diff
@@ -12,10 +13,49 @@
...
+        n = 0
+        while True:
+            a = factor * base ** n
+            if n >= tries:
+                yield remainder
+            else:
+                yield a
+                n += 1
+
+    # tries is the tells us the largest standard wait using the standard progression (before being capped)
```
Comment (doctoryes) [inline tubular/jenkins.py]: Nit: Typo.

File: tubular/jenkins.py [hunk 3]
```diff
@@ -12,10 +13,49 @@
...
+
+def _backoff_timeout(timeout, base=2, factor=1):
+    u"""Return a tuple of (wait_gen, max_tries) so that backoff will only try up to `timeout` seconds."""
+    # Total duration of sum(factor * base ** n for n in range(K)) = factor*(base**K - 1)/(base - 1)
+    #
+    # Solving for K, K = log(timeout * (base - 1) / factor + 1, base)
+    #
+    # Using the next smallest integer K will give us a number of elements from
+    # the exponential sequence to take and still be less than the timeout.
+    tries = int(math.log(timeout * (base - 1) / factor + 1, base))
```
Comment (doctoryes) [inline tubular/jenkins.py:55]: A timeout of 0 or base of 1 makes this equation blow up, though arguably it should.

File: tubular/jenkins.py [hunk 4]
```diff
@@ -30,52 +70,27 @@ def trigger_build(base_url, user_name, user_token, job_name, job_token, job_caus
...
-        |4    |4   |7    |0.12  |
-        |8    |5   |15   |0.25  |
-        |16   |6   |31   |0.52  |
-        |32   |7   |63   |1.05  |
-        |64   |8   |127  |2.12  |
-        |128  |9   |255  |4.25  |
-        |256  |10  |511  |8.52  |
-        |512  |11  |1023 |17.05 |
-        |1024 |12  |2047 |34.12 |
-        |2048 |13  |4095 |68.25 |
```
Comment (doctoryes) [inline tubular/jenkins.py:78]: Nit: I understand that the new backoff is different - but this shorthand table was useful in getting a sense of how often the polls were happening. Maybe leave something similar to this table in the code?

File: tubular/jenkins.py (hunk 3 above)
Comment (jzoldak) [reply tubular/jenkins.py:55]: Can you pls add tests for this method (even tho it's private)? I think it might be off by one?

```
for x in [1, 3, 7, 14, 15, 16, 30, 31, 32]:
    foo, bar =_backoff_timeout(x)
    print 'timeout: {}, tries: {}'.format(x,bar)
```
gives this output:
```
timeout: 1, tries: 3
timeout: 3, tries: 4
timeout: 7, tries: 5
timeout: 14, tries: 5
timeout: 15, tries: 6
timeout: 16, tries: 6
timeout: 30, tries: 6
timeout: 31, tries: 7
timeout: 32, tries: 7
```
Which doesn't seem to match up to the original table (not to say that I didn't make an arithmetic mistake):
```
        |wait x sec|make attempt #|total time (sec)|total time (min)|
        |----:|---:|----:|-----:|
        |0    |1   |0    | 0    |
        |1    |2   |1    |0.02  |
        |2    |3   |3    |0.05  |
        |4    |4   |7    |0.12  |
        |8    |5   |15   |0.25  |
        |16   |6   |31   |0.52  |
        |32   |7   |63   |1.05  |
        |64   |8   |127  |2.12  |
        |128  |9   |255  |4.25  |
        |256  |10  |511  |8.52  |
        |512  |11  |1023 |17.05 |
        |1024 |12  |2047 |34.12 |
        |2048 |13  |4095 |68.25 |
```

File: tubular/jenkins.py (hunk 4 above)
Comment (cpennington) [reply tubular/jenkins.py:78]: Sure.

File: tubular/jenkins.py (hunk 3 above)
Comment (cpennington) [reply tubular/jenkins.py:55]: `_backoff_timeout` takes a timeout in seconds, not minutes, which I think explains the inconsistencies with your table.

Comment (cpennington) [reply tubular/jenkins.py:55]: Ah, nevermind. There's an off-by-one error, it appears.

File: tubular/jenkins.py (hunk 4 above)
Comment (cpennington) [reply tubular/jenkins.py:78]: Ok, readded.

File: tubular/jenkins.py (hunk 3 above)
Comment (cpennington) [reply tubular/jenkins.py:55]: Fixed it.

**Review by doctoryes — APPROVED** [review]
